components/dashboard/src/components/RepositoryFinder.tsx

```diff
@@ -0,0 +1,159 @@
+        }
+    }
+
+    // Up/Down keyboard navigation between results
+    const onKeyDown = (event: React.KeyboardEvent) => {
+        if (!selectedSearchResult) {
+            return;
+        }
+        const selectedIndex = searchResults.indexOf(selectedSearchResult);
+        const select = (index: number) => {
+            // Implement a true modulus in order to "wrap around" (e.g. `select(-1)` should select the last result)
+            // Source: https://stackoverflow.com/a/4467559/3461173
+            const n = Math.min(searchResults.length, MAX_DISPLAYED_ITEMS);
+            setSelectedSearchResult(searchResults[((index % n) + n) % n]);
+        }
+        if (event.key === 'ArrowDown') {
+            event.preventDefault();
+            select(selectedIndex + 1);
+            return;
+        }
+        if (event.key === 'ArrowUp') {
+            event.preventDefault();
+            select(selectedIndex - 1);
+            return;
+        }
+    }
+
+    useEffect(() => {
+        const element = document.querySelector(`a[href='/#${selectedSearchResult}']`);
+        if (element) {
+            element.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+        }
+    }, [selectedSearchResult]);
+
+    const onSubmit = (event: React.FormEvent) => {
+        event.preventDefault();
+        if (selectedSearchResult) {
+            window.location.href = '/#' + selectedSearchResult;
+        }
+    }
+
+    return <form onSubmit={onSubmit}>
+        <div className="flex px-4 rounded-xl border border-gray-300 dark:border-gray-500">
+            <div className="py-4">
+                <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 16 16" width="16" height="16"><path fill="#A8A29E" d="M6 2a4 4 0 100 8 4 4 0 000-8zM0 6a6 6 0 1110.89 3.477l4.817 4.816a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 010 6z" /></svg>
+            </div>
+            <input type="search" className="flex-grow" placeholder="Repository" autoFocus value={searchQuery} onChange={e => search(e.target.value)} onKeyDown={onKeyDown} />
+        </div>
+        <div className="mt-3 flex flex-col space-y-2 h-64 overflow-y-auto" id="search-results">
+            {searchResults.slice(0, MAX_DISPLAYED_ITEMS).map((result, index) =>
+                <a className={`px-4 py-3 rounded-xl` + (result === selectedSearchResult ? ' bg-gray-600 text-gray-50 dark:bg-gray-700' : '')} href={`/#${result}`} key={`search-result-${index}`} onMouseEnter={() => setSelectedSearchResult(result)}>
+                    {result.split(searchQuery).map((segment, index) => <span>
+                        {index === 0 ? <></> : <strong>{searchQuery}</strong>}
+                        {segment}
+                    </span>)}
+                </a>
+            )}
+            {searchResults.length > MAX_DISPLAYED_ITEMS &&
+                <span className="mt-3 px-4 py-2 italic text-sm">{searchResults.length - MAX_DISPLAYED_ITEMS} results not shown</span>}
```

Comment:
**issue**: Why the limit here? I thought we decided to go with the scrollbar because then we could always show all results. Otherwise, there's not much different with the fixed result set we discussed earlier this week, right? Cc @jldec 

**suggestion**: In any case, going with a fixed max number of results and infiniter scroll pagination would be ideal, but sounds also ok to leave out of the scope of this PR. 💭 

<img width="169" alt="Screenshot 2022-01-21 at 6 11 21 PM" src="https://user-images.githubusercontent.com/120486/150563238-e2beab61-b5d8-41bd-a8ed-3a15702e284f.png">

Reply:
For performance reasons. I noticed the UI started struggling a bit at around 25 results. Also, I found that anything after 10-15 results is no longer relevant for my query.

Happy to adjust as needed, but there should probably always be a hard limit for performance reasons (also, Jurgen's concern was about seeing 3 elements and wanting to see more, I guess when you see 10-15 results you no longer have this problem).

Reply:
Ah, I see! Ideally, we could paginate the results with an infinite scroll but sounds good to leave this out of the scope of this PR. However, there's definitely some value in surfacing the amount of results found.

**suggestion**: What do you think of rephrasing this to the following?

```suggestion
                <span className="mt-3 px-4 py-2 text-sm text-gray-400 dark:text-gray-500">{searchResults.length - MAX_DISPLAYED_ITEMS} results found</span>}
```

| BEFORE | AFTER | 
|-|-|
| <img width="426" alt="results-before" src="https://user-images.githubusercontent.com/120486/152229949-86e2e89f-5922-4511-846f-43182c4fdfe9.png"> | <img width="426" alt="results-after" src="https://user-images.githubusercontent.com/120486/152229952-95112207-ddc0-42da-a139-aa6a2db8ab28.png"> |

Ideally this could also use singular when only 1 more result has been found. However, feel free to leave this as is for now as more improvements could come into this area.

Reply:
Thanks @gtsiolis!

The goal of this was to give users a hint that "there are more items beyond this point" -- we actually expect the most relevant results to be among the first 15-20 items, but in some edge cases users might want to know if there are even more matches.

What do you think of an alternative phrasing like this one?

```
Showing 15/42 matches
```

Reply:
Ah, I've just noticed the "2 more results found" text in https://github.com/gitpod-io/gitpod/pull/7715#discussion_r797990648

Using "{N} more results found" does work for me. 👍

Reply:
Thanks for tackling both comments, @jankeromnes! The designs in the comment mentioned in https://github.com/gitpod-io/gitpod/pull/7715#discussion_r800817010 are definitely more preferable.

Going with `{N} more results found` sounds perfect. 🏀 

Cross-posting from https://github.com/gitpod-io/gitpod/pull/7715#discussion_r797972968 for visibility and future reference:

> Ideally, we could paginate the results with an infinite scroll but sounds good to leave this out of the scope of this PR. However, there's definitely some value in surfacing the amount of results found.